src/Helpers/Traits/WaitForWakeTrait.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Pantheon\Terminus\Helpers\Traits;
+
+use Pantheon\Terminus\Exceptions\TerminusException;
+use Pantheon\Terminus\Models\Environment;
+use Psr\Log\LoggerInterface;
+
+trait WaitForWakeTrait
+{
+    /**
+     * Waits for the site to wake up.
+     *
+     * @param Environment $env
+     * @throws \Exception
+     */
+    public function waitForWake(Environment $env, LoggerInterface $logger)
+    {
+        $waits = 0;
+        do {
+            $woke = $env->wake();
+            if (($woke['success'] ?? false) === true) {
+                break;
+            }
+            // if success is empty, then the site is still waking up.
+            // If we've waited more than 25 seconds, then something is wrong.
+            if ($waits > 5) {
```

Comment:
Looks great, except I think we should let this spin for 25 sec or so, like the comment says. Now that the sleep is only 1s, this delay is only 5s.